pkg/grpc/api.go

```diff
@@ -658,7 +658,7 @@ func (a *api) SaveState(ctx context.Context, in *runtimev1pb.SaveStateRequest) (
 		}
 
 		if s.Etag != nil {
-			req.ETag = &s.Etag.Value
+			req.ETag = &in.States[i].Etag.Value
```

Comment:
Is it a fix for etag?

Reply:
This is to stop the Go linter from complaining

```
 G601: Implicit memory aliasing in for loop.
```